editor/script/paint.js

```diff
@@ -327,32 +327,53 @@ function PaintTool(canvas, roomTool) {
 		return self.drawing.getEngineObject();
 	}
 
-	this.newDrawing = function() {
+	this.newDrawing = function(imageData) {
 		if ( self.drawing.type == TileType.Tile ) {
-			newTile();
+			newTile(undefined, imageData);
 		}
 		else if( self.drawing.type == TileType.Avatar || self.drawing.type == TileType.Sprite ) {
-			newSprite();
+			newSprite(undefined, imageData);
 		}
 		else if( self.drawing.type == TileType.Item ) {
-			newItem();
+			newItem(undefined, imageData);
 		}
 
 		// update paint explorer
 		self.explorer.AddThumbnail( self.drawing.id );
 		self.explorer.ChangeSelection( self.drawing.id );
 		document.getElementById("paintExplorerFilterInput").value = ""; // super hacky
 		self.explorer.Refresh( self.drawing.type, true /*doKeepOldThumbnails*/, document.getElementById("paintExplorerFilterInput").value /*filterString*/, true /*skipRenderStep*/ ); // this is a bit hacky feeling
-	}
+    }
+    
+    this.duplicateDrawing = function() {
+        let objectId = getIdPrefix(self.drawing.type) + self.drawing.id;
```

Comment:
I want to get rid of the editor-level drawing object someday, but in the meantime you can use `self.drawing.toString()` to get the type + id string